packages/arb-validator-core/message/transactionBatch.go

```diff
@@ -0,0 +1,315 @@
+		len(m.Data) != len(o.Data) ||
+		len(m.Signatures) != len(o.Signatures) {
+		return false
+	}
+
+	for i := range m.Tos {
+		if m.Tos[i] != o.Tos[i] {
+			return false
+		}
+	}
+	for i := range m.Tos {
+		if m.SequenceNums[i].Cmp(o.SequenceNums[i]) != 0 {
+			return false
+		}
+	}
+	for i := range m.Tos {
+		if m.Values[i].Cmp(o.Values[i]) != 0 {
+			return false
+		}
+	}
+	if !bytes.Equal(m.Data, o.Data) {
+		return false
+	}
+	if !bytes.Equal(m.Signatures, o.Signatures) {
+		return false
+	}
+
+	return true
+}
+
+func (m TransactionBatch) Type() MessageType {
+	return TransactionBatchType
+}
+
+type DeliveredTransactionBatch struct {
+	TransactionBatch
+	BlockNum  *common.TimeBlocks
+	Timestamp *big.Int
+}
+
+func (m DeliveredTransactionBatch) Equals(other Message) bool {
+	o, ok := other.(DeliveredTransactionBatch)
+	if !ok {
+		return false
+	}
+	return m.TransactionBatch.Equals(o.TransactionBatch) &&
+		m.BlockNum.Cmp(o.BlockNum) == 0 &&
+		m.Timestamp.Cmp(o.Timestamp) == 0
+}
+
+func (m DeliveredTransactionBatch) deliveredHeight() *common.TimeBlocks {
+	return m.BlockNum
+}
+
+func (m DeliveredTransactionBatch) deliveredTimestamp() *big.Int {
+	return m.Timestamp
+}
+
+func (m DeliveredTransactionBatch) CommitmentHash() common.Hash {
+	addressTy, _ := abi.NewType("address", "", nil)
+	addressArrayTy, _ := abi.NewType("address[]", "", nil)
+	uint256ArrayTy, _ := abi.NewType("uint256[]", "", nil)
+	uint32ArrayTy, _ := abi.NewType("uint32[]", "", nil)
+	bytesTy, _ := abi.NewType("bytes", "", nil)
+	args := abi.Arguments{
+		abi.Argument{
+			Type: addressTy,
+		},
+		abi.Argument{
+			Type: addressArrayTy,
+		},
+		abi.Argument{
+			Type: uint256ArrayTy,
+		},
+		abi.Argument{
+			Type: uint256ArrayTy,
+		},
+		abi.Argument{
+			Type: uint32ArrayTy,
+		},
+		abi.Argument{
+			Type: bytesTy,
+		},
+		abi.Argument{
+			Type: bytesTy,
+		},
+	}
+	packedData, err := args.Pack(
+		m.Chain,
+		common.AddressArrayToEth(m.Tos),
+		m.SequenceNums,
+		m.Values,
+		m.DataLengths,
+		m.Data,
+		m.Signatures,
+	)
+	if err != nil {
+		log.Fatal(err)
+	}
+
+	return hashing.SoliditySHA3(
+		hashing.Uint8(uint8(m.Type())),
+		packedData,
+		hashing.Uint256(m.BlockNum.AsInt()),
+		hashing.Uint256(m.Timestamp),
+	)
+}
+
+func (m DeliveredTransactionBatch) CheckpointValue() value.Value {
+	innerTup, _ := value.NewTupleFromSlice([]value.Value{
+		value.NewIntValue(new(big.Int).Set(m.BlockNum.AsInt())),
+		value.NewIntValue(new(big.Int).Set(m.Timestamp)),
+	})
+	val, _ := value.NewTupleFromSlice([]value.Value{
+		addressToIntValue(m.Chain),
+		addressesToValue(m.Tos),
+		intsToValue(m.SequenceNums),
+		intsToValue(m.Values),
+		uint32sToValue(m.DataLengths),
+		BytesToByteStack(m.Data),
+		BytesToByteStack(m.Signatures),
+		innerTup,
+	})
+	return val
+}
+
+func UnmarshalTransactionBatchFromCheckpoint(v value.Value) (DeliveredTransactionBatch, error) {
+	tup, ok := v.(value.TupleValue)
+	failRet := DeliveredTransactionBatch{}
+	if !ok || tup.Len() != 8 {
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [avoid magic numbers like '8', create a named constant for it](https://app.codacy.com/gh/OffchainLabs/arbitrum/pullRequest?prid=5428340)